Lib/test/support/os_helper.py

```diff
@@ -508,9 +535,36 @@ def temp_dir(path=None, quiet=False):
             rmtree(path)
 
 
+@contextlib.contextmanager
+def save_cwd(*, quiet=False):
```

Comment:
This new function may not be needed in light of `change_cwd` (see the other comment that uses it).